pkg/security/probe/probe_ebpf.go

```diff
@@ -886,33 +886,25 @@ func (p *EBPFProbe) handleEvent(CPU int, data []byte) {
 			return
 		}
 
-		if process.IsKThread(event.ProcessCacheEntry.PPid, event.ProcessCacheEntry.Pid) {
-			return
-		}
-
-		p.Resolvers.ProcessResolver.ApplyBootTime(event.ProcessCacheEntry)
-		event.ProcessCacheEntry.SetSpan(event.SpanContext.SpanID, event.SpanContext.TraceID)
-
-		p.Resolvers.ProcessResolver.AddForkEntry(event.ProcessCacheEntry, event.PIDContext.ExecInode, newEntryCb)
+		p.Resolvers.ProcessResolver.AddForkEntry(event, newEntryCb)
 	case model.ExecEventType:
 		// unmarshal and fill event.processCacheEntry
 		if _, err = p.unmarshalProcessCacheEntry(event, data[offset:]); err != nil {
 			seclog.Errorf("failed to decode exec event: %s (offset %d, len %d)", err, offset, len(data))
 			return
 		}
 
-		if err = p.Resolvers.ProcessResolver.ResolveNewProcessCacheEntry(event.ProcessCacheEntry, event.ContainerContext); err != nil {
-			seclog.Debugf("failed to resolve new process cache entry context for pid %d: %s", event.PIDContext.Pid, err)
-
+		err = p.Resolvers.ProcessResolver.AddExecEntry(event)
+		if err != nil {
 			var errResolution *path.ErrPathResolution
 			if errors.As(err, &errResolution) {
```

Comment:
We should move this error handling withing the AddExecEntry function otherwise we are going to change the current behavior which is

Path error => set the event as in error & insert the entry

With your changes, if there is an path resolution error, the entry isn't inserted